clang/lib/Sema/SemaBounds.cpp

```diff
@@ -1858,7 +1873,7 @@ namespace {
            if (NestedElements.find(S) != NestedElements.end())
              continue;
 
-           CheckedScopeSpecifier CSS = CheckedScopeSpecifier::CSS_Unchecked;
+           CSS = CheckedScopeSpecifier::CSS_Unchecked;
```

Comment:
I think it is confusing to directly set the CSS member here.  I'd recommend using the RAII pattern that is now being used elsewhere in the file.

Reply:
Based on our discussion, I made CSS a parameter passed through CheckBoundsDeclarations methods again.